frontend/src/components/ProfileContent.tsx

```diff
@@ -51,78 +75,259 @@ const useStyles = makeStyles({
-      value: props.religion,
+      title: "Religions",
+      value: readonlyContent.religions.join(", "),
+      textFieldProps: {
+        required: false,
+        disabled: false,
+        onChange: (e) => {
+          setters.setReligions(e.currentTarget.value.split(","))
+        },
+      },
     },
   ]
 
+  const handleEditBtnClick = () => {
+    toggleEditMode(true)
+  }
+
+  const handleCancelBtnClick = () => {
+    seterrMessage("")
+    toggleEditMode(false)
+  }
+
+  const saveFields = () => {
+    React.useEffect(() => {
+      request<SaveProfileResponse>("/api/current_user/", "post", true, true)
+        .then(() => {
+          setProfile(editableContent)
+          toggleEditMode(false)
+          seterrMessage("")
+        })
+        .catch((err) => {
+          seterrMessage(err)
+        })
+    })
+  }
+
   return (
-    <>
-      <Grid container item className={classes.profileHeader} xs={12}>
-        <Grid container item alignItems="center" justify="center" xs={6}>
-          <Avatar
-            alt={props.name}
-            src={props.profileImg}
-            className={classes.profileAvatar}
-          />
-        </Grid>
-        <Grid container item alignItems="center" justify="center" xs={6}>
-          <div className={classes.introVideo}>Intro Video</div>
-        </Grid>
-      </Grid>
-      <Grid
-        container
-        item
-        alignItems="center"
-        className={classes.fieldsContainer}
-        xs={12}
-      >
-        {profileFields.map(({ title, value }: ProfileField) => (
-          <Grid key={title} item className={classes.field} sm={12} md={6}>
-            <Typography>
-              {title}: {value}
-            </Typography>
+    <Grid container spacing={2}>
+      <Grid item className={classes.profileHeader} xs={12}>
+        <Grid container justify="center" spacing={2}>
+          <Grid item xs={12} sm={6}>
+            <Avatar
+              src={readonlyContent.profileImg}
+              className={classes.profileAvatar}
+            />
+          </Grid>
+          <Grid item xs={12} sm={6}>
+            <div className={classes.introVideo}>Intro Video</div>
           </Grid>
-        ))}
+        </Grid>
       </Grid>
-    </>
+      {!isEditMode && (
+        <Grid item xs={12}>
+          <Paper>
+            <Grid container spacing={2}>
+              {fields.map(({ title, value }: ProfileField) => (
+                <Grid
+                  key={`readonlyField-${title}`}
+                  container
+                  item
+                  justify="flex-end"
+                  className={classes.field}
+                  sm={12}
+                  md={6}
+                >
+                  <Grid item xs={9}>
+                    <Typography>
+                      {title}: {value}
+                    </Typography>
+                  </Grid>
+                </Grid>
+              ))}
+              <Grid container item justify="center" xs={12}>
```

Comment:
I believe this throws an error if a Grid is both a container and an item

Reply:
This is fine: https://material-ui.com/components/grid/#nested-grid

Reply:
Weird, I got warnings when I tried it

Reply:
What are the warnings? I don't have warnings.

Reply:
I'll look into another time